### PR TITLE
Inclusion of token authenticity

### PR DESCRIPTION
**Pull request by gklps**

### CODE REVIEW

File: core/token_auth.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+func (c *Core) getWholeTokenValue(wholeTokenName string) (int, string) {
```
Comment (muraliens) [inline core/token_auth.go:11]: Add this as separate package outside core, like RAC create separate package token and add this files,

File: core/token_auth.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	response.Close()
+
+	tokenLevel := string(trimmedResult[:len(trimmedResult)-64])
+	tokenCountHash := string(trimmedResult[len(trimmedResult)-64:])
+
+	fmt.Println(len(tokenCountHash), "token len")
+
+	tokenLevelInt, err := strconv.Atoi(tokenLevel)
+	if err != nil {
+		fmt.Println("Error occured", "error", err)
```
Comment (muraliens) [inline core/token_auth.go:36]: handle the errors properly

**Comment by gklps** [comment]
@muraliens Made the changes as per your comments

File: core/token_auth.go (hunk 1 above)
Comment (gklps) [reply core/token_auth.go]: moved to a separate package called token

File: core/token_auth.go (hunk 2 above)
Comment (gklps) [reply core/token_auth.go]: updated

**Review by muraliens — APPROVED** [review]

**Pull request merged**